src/main/java/org/craftercms/studio/impl/v2/service/content/internal/ContentServiceInternalImpl.java

```diff
@@ -208,6 +219,59 @@ public DetailedItem getItemById(String siteId, long id, boolean preferContent)
         return detailedItem;
     }
 
+    @Override
+    public List<SandboxItem> getSanboxItemsByPath(String siteId, List<String> paths, boolean preferContent)
+            throws ServiceLayerException, UserNotFoundException {
+        Map<String, String> params = new HashMap<String, String>();
+        params.put(SITE_ID, siteId);
+        SiteFeed siteFeed = siteFeedMapper.getSite(params);
+        List<SandboxItem> toRet = new ArrayList<SandboxItem>();
+        List<Item> items = null;
+        if (preferContent) {
+            items = itemDao.getSandboxItemsByPathPreferContent(siteFeed.getId(), paths);
+        } else {
+            items = itemDao.getSandboxItemsByPath(siteFeed.getId(), paths);
+        }
+        if (CollectionUtils.isNotEmpty(items)) {
```

Comment:
Seems that the filtering and available actions are the same for byPath and byId, please factor this section out into a common method so we have one place for that code to be maintained.